Objective-C/CBLReplicator.mm

```diff
@@ -358,6 +358,67 @@ - (void) removeChangeListenerWithToken: (id<CBLListenerToken>)token {
     }
 }
 
+- (NSSet<NSString*>*) pendingDocumentIds: (NSError**)error {
+    if (_config.replicatorType > 1) {
+        if (error)
+            *error = [NSError errorWithDomain: CBLErrorDomain
+                                         code: CBLErrorUnsupported
+                                     userInfo: @{NSLocalizedDescriptionKey: kCBLErrorMessagePullOnlyPendingDocIDs}];
+        return nil;
+    }
+    
+    C4Error c4err = {};
+    if (![self _setupC4Replicator: &c4err]) {
+        convertError(c4err, error);
+        CBLWarnError(Sync, @"%@: Replicator cannot be created: %d/%d", self, c4err.domain, c4err.code);
+        return nil;
+    }
+    
+    C4SliceResult result = c4repl_getPendingDocIDs(_repl, &c4err);
```

Comment:
Question: Does the c4repl_getPendingDocIDs need to be called under a database's lock?

Reply:
Assuming this API is thread safe, able to see a mutex lock in the definition of API [here](https://github.com/couchbase/couchbase-lite-core/blob/c6f8346ea925b7ba60200ced92e3b6770d34d54c/Replicator/c4Replicator.hh#L124)